### PR TITLE
fix: resolve error on close items

### PR DESCRIPTION
**Pull request by Sebastien-Ahkrin**
close: https://github.com/zakodium-oss/analysis-ui-components/issues/372

### CODE REVIEW

**Comment by targos** [comment]
Please think about your commit messages: they should explain what you do. Here there was no error. I reworded to `fix: do not close DropdownMenu when disabled items are clicked`

**Comment by targos** [comment]
The problem now is that it also doesn't close when you click on enabled items...

**Comment by Sebastien-Ahkrin** [comment]
> The problem now is that it also doesn't close when you click on enabled items...

Same as Dropdown with context.
In fact, the thing that make the menu close was a bug (with the clickOutside), so do we have to call the close function on both of the components ?

**Comment by targos** [comment]
What is Dropdown with context?

**Comment by Sebastien-Ahkrin** [comment]
> What is Dropdown with context?

the one with the right click

**Comment by targos** [comment]
Yes, we have to call the close function in both.

File: src/components/dropdown-menu/MenuItems.tsx [hunk 1]
```diff
@@ -126,7 +126,11 @@ function Item<T>(props: ItemProps<T>) {
     <Menu.Item disabled={option.disabled}>
       {({ active }) => (
         <ItemDiv
-          onClick={() => onSelect(option)}
+          onClick={() => {
+            if (!option.disabled) {
+              onSelect(option);
+            }
+          }}
```
Comment (targos) [inline src/components/dropdown-menu/MenuItems.tsx:133]: should we remove this now that you call it above?

Comment (targos) [reply src/components/dropdown-menu/MenuItems.tsx]: I mean remove everything. You already call `onSelect` above.

Comment (targos) [reply src/components/dropdown-menu/MenuItems.tsx]: @Sebastien-Ahkrin

Comment (Sebastien-Ahkrin) [reply src/components/dropdown-menu/MenuItems.tsx]: i'm never calling onSelect if i remove theses line

**Pull request merged**